File: 백준/Silver/11047. 동전 0/동전 0.py
n, k = map(int, input().split())

lst = [int(input()) for _ in range(n)]
lst.sort(reverse=True)
cnt = 0
for i in range(n):
    if lst[i] <= k:
        mok = k //lst[i]
        cnt += mok
        k -= lst[i]*mok
print(cnt)

File: 백준/Silver/11047. 동전 0/동전 0.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const [N, K] = input.shift().split(" ").map(Number);
const lst = input.map(Number);
let money = K;
let coin = 0;

for (let i = lst.length - 1; i >= 0; i--) {
  if (money >= lst[i]) {
    coin += Math.floor(money / lst[i]);
    money = money % lst[i];
  }
  if (money === 0) {
    break;
  }
}

console.log(coin);
